colour/continuous/signal.py

```diff
@@ -515,6 +517,7 @@ def extrapolator_kwargs(self, value: dict):
         self._function = None  # Invalidate the underlying continuous function.
 
     @property
+    @ndarray_copy_enable(False)
```

Comment:
It's not clear to me why these decorators are littered around... I can't see where they are actually affecting any behavior. It looks like it is a promise that this function is copy-less safe (no-side effects) but that guarantee or promise didn't exist before and doesn't appear to be adding value to the function signature here... Are we going to end up with `@nd_copy_enable(False)` functions everywhere? 

Minor concern about long term code litter.

Reply:
This is a good concern, and one would certainly have to know what is going on here, i.e. that it disable the `domain` and `range` getter copies. We could add some comments to make it understandable. I don't think there is a need to have the decorators anywhere else than where they are.

An alternative is to have getters that are without copy, e.g. `domain_no_copy` or alike.

Reply:
I just did a small pass using directly the protected attributes thus removed most of the decorated calls.